### DISCUSSION TITLE
python3-pykdl] defined as "not available" for OS version [bionic]

### DISCUSSION THREAD

**Discussion by islemhm** [prl-mushr/mushr#29] [General]
hello ihave a problem when i run ''''rosdep install -y --from-paths . --ignore-src --rosdistro melodic''' it show me this error """"ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
tf2_geometry_msgs: [python3-pykdl] defined as "not available" for OS version [bionic]
tf2_sensor_msgs: [python3-pykdl] defined as "not available" for OS version [bionic]
amcl: [python3-pykdl] defined as "not available" for OS version [bionic]
"""" any one can help me plzz i use ubuntu 18.04 and ros melodic

**Comment by schmittlema** [comment]
Hi @islemhm! What package are you trying to install dependencies for? This command I don't think is used anywhere in our tutorials, so it may not work as expected. That being said, it seems your default python version it set to python3 which is not compatible with melodic. The installer is trying to find python3-pykdl which does not exist for bionic. There is a python2 version that it should be trying to install, but for some reason it thinks it should install the python3 version.

**Reply by islemhm** [reply]
@schmittlema   i tried to install moveit i follow this tutoriel " http://docs.ros.org/en/melodic/api/moveit_tutorials/html/doc/getting_started/getting_started.html 
how i can remove python3 and install python 2

**Reply by schmittlema** [reply]
I'm not sure exactly how to fix this as this is not mushr specific. I would try installing tf2 manually
`sudo apt install ros-melodic-tf2-geometry-msgs ros-meloditc-tf2-sensor-msgs`

**Reply by islemhm** [reply]
@schmittlema  thank you 
that did not solve the problem

**Reply by palanc** [reply]
Hi @islemhm - I like this tutorial for switching between Python 2 & 3: https://www.fosslinux.com/39384/switching-between-python-2-and-3-versions-on-ubuntu-20-04.htm

If that doesn't help, could you tell us the output of the following two commands:

**ls /usr/bin/python***

**sudo update-alternatives --list python**

**Reply by islemhm** [reply]
@planc509  
 ls /usr/bin/python*
/usr/bin/python
/usr/bin/python2
/usr/bin/python2.7
/usr/bin/python2.7-config
/usr/bin/python2-config
/usr/bin/python2-qr
/usr/bin/python3
/usr/bin/python3.6
/usr/bin/python3.6-config
/usr/bin/python3.6-coverage
/usr/bin/python3.6m
/usr/bin/python3.6m-config
/usr/bin/python3-config
/usr/bin/python3-coverage
/usr/bin/python3m
/usr/bin/python3m-config
/usr/bin/python3-unidiff
/usr/bin/python-argcomplete-check-easy-install-script3
/usr/bin/python-argcomplete-tcsh3
/usr/bin/python-config



 sudo update-alternatives --list python
/usr/bin/python3

**Reply by palanc** [reply]
I see, could you also post a screenshot of the output of this command:

**sudo update-alternatives --config python**

**Reply by islemhm** [reply]
@planc509 
![image](https://user-images.githubusercontent.com/79261716/121731233-91118500-cae8-11eb-94a9-c26a168f74c3.png)

**Reply by palanc** [reply]
Great if you run that command again and then enter a '1' when it asks you to make a selection, that should switch your default python version to be python 2. Does that fix your original problem?

**Reply by islemhm** [reply]
@planc509  no it doesn't fix it, it still the same problem

**Reply by palanc** [reply]
Are you still getting the _exact_ same error message as in your original post?

Also, what is the output of this command:

**python --version**

**Reply by islemhm** [reply]
@planc509   when i run :python --version
Python 2.7.17
yes it still the same error msg

**Reply by palanc** [reply]
Sorry I'm not sure why it is still referencing python3-dkl if you are using python 2 now. Unless anyone else here has ideas, maybe you could try contacting the maintainers of moveit to see if they have seen this problem before?

**Reply by islemhm** [reply]
@planc509 thank you for your help
